src/lib.rs

```diff
@@ -168,6 +170,30 @@ pub trait Arbitrary: Sized + 'static {
     /// See also the documentation for [`Unstructured`][crate::Unstructured].
     fn arbitrary(u: &mut Unstructured<'_>) -> Result<Self>;
 
+    /// Get a size hint for how many bytes out of an `Unstructured` this type
+    /// needs to construct itself.
+    ///
+    /// This is useful for determining how many elements we should insert when
+    /// creating an arbitrary collection.
+    ///
+    /// The return value is similar to
+    /// [`Iterator::size_hint`][iterator-size-hint]: it returns a tuple where
+    /// the first element is a lower bound on the number of bytes required, and
+    /// the second element is an optional upper bound.
```

Comment:
Should this be an optional upper bound, or an exact value? Will there be cases where having an upper bound is more useful than an exact value?

Reply:
I don't think we support any types in this crate where that distinction matters so this just would involve changing the comment, but making it an exact value simplifies the next steps for the container stuff, I _think_

Reply:
It actually might not matter at all. Your call.

Reply:
Also I just read your comment on the issue. Never mind :)